homeassistant/components/zha/helpers.py

```diff
@@ -1013,7 +1013,11 @@ def async_get_zha_device_proxy(hass: HomeAssistant, device_id: str) -> ZHADevice
         raise KeyError(f"Device id `{device_id}` not found in registry.")
     zha_gateway_proxy = get_zha_gateway_proxy(hass)
     try:
-        ieee_address = list(registry_device.identifiers)[0][1]
+        ieee_address = next(
```

Comment:
This will now throw a `StopIteration` error if no matching identifiers can be found. This won't be caught by the `except (IndexError, ValueError) as ex:` below.